prov/util/src/util_buf.c

```diff
@@ -192,6 +192,9 @@ int ofi_bufpool_create_attr(struct ofi_bufpool_attr *attr,
 	if (pool->attr.flags & OFI_BUFPOOL_HUGEPAGES) {
 		pool->alloc_size = ofi_get_aligned_size(pool->alloc_size,
 							hp_size);
+	} else if (attr->alignment) {
+		pool->alloc_size = ofi_get_aligned_size(pool->alloc_size,
+							attr->alignment);
```

Comment:
> What is this change fixing?

It is trying to fix a potential data corruption, cause by rdma-core's fork support. When rdma-core's fork support is enabled. It will call `madvise(MADV_DONTFORK)` on  the registered area, because `madvise` requires its input to be page aligned, rdma-core will extend the region to be madvised to cover the who page, e.g. madvise will be called on a memory that is not registered. 

The memory region that is advised as `MADV_DONTFORK` is not visible to the child process. So if the extended region happen to have information that child need, the child will have corruption. We have not seen this corruption happen with the buf pool memory, but still want to make the pool memory region to be page aligned.

Reply:
I have now confirmed that this does indeed cause memory corruption.  

As for the question of why adjust allocation size: I would argue that if the caller requests an aligned allocation they would like to avoid sharing that allocation unit with anyone else, be it a cache line or a page.  Aligning only the start of the allocation but not its size just leads to a nasty surprise when some unrelated caller of `malloc` down the road starts sharing that allocation unit.

Reply:
The reason for the change just needs to be captured in the commit message, so it's clear why it was made.  I would not assume anything about the caller's need for an aligned allocation.  They could have 128-bit atomic values that need to be aligned for all this code knows.

As mentioned above, if we increase the allocation size, then we should also increase the chunk_cnt to account for the extra space.  Otherwise, we're just wasting the extra allocated space.

Reply:
Taking a closer look I see that this part of the change is unnecessary: we are already padding `entry_size` to the correct alignment, and a multiple of `entry_size` will still meet alignment requirements.  This line is a no-op so I'll remove it.

Reply:
On second thought, we don't want to pad each entry size, but instead align the whole pool.  I'll revise.